docs/source/framework.rst

```diff
@@ -174,6 +174,39 @@ This provide a base class that will behave as normal :code:`DataContainer`.
    :members:
    :inherited-members:
 
+Partitioner
+===========
+
+This class allows a user to partition an instance of :code:`AcquisitionData` into a number of subsets. For example, to use with a stochastic optimisation method. 
+
+For example: 
+
+.. code-block :: python
+
+   from cil.utilities import dataexample
+   from cil.plugins.astra.operators import ProjectionOperator
+   
+   # get the data  
+   data = dataexample.SIMULATED_PARALLEL_BEAM_DATA.get()
+   data.reorder('astra')
+   data = data.get_slice(vertical='centre')
+
+   # create the geometries 
+   ag = data.geometry 
+   ig = ag.get_ImageGeometry()
+
+   # partition the data and build the projectors
+   n_batches = 10 
+   partitioned_data = data.partition(num_batches=n_batches, mode='sequential') # Choose mode from `sequential`, `staggered` or `random_permutation` 
+   A_partitioned = ProjectionOperator(ig, partitioned_data.geometry, device = "cpu")
+
```

Comment:
Can the example do something beyond just defining to illustrate these objects, that would help clarity. Maybe just print a list of them, or apply direct or adjoint to something to demonstrate the machinery

Reply:
Very nice!